src/js/pages/jobs/JobsTab.js

```diff
@@ -76,185 +127,56 @@ class JobsTab extends mixin(StoreMixin) {
-      delete query[filterKey];
-    });
-
-    this.context.router.push({ pathname, query });
-  }
-
-  resetFilter() {
-    var state = Object.assign({}, this.state, DEFAULT_FILTER_OPTIONS);
-    this.setState(state, this.resetFilterQueryParams);
-  }
-
-  updateFromProps(props) {
-    const { location = { query: {} } } = props;
-
-    if (location.query[JobFilterTypes.TEXT] != null) {
-      const state = {};
-
-      state[JobFilterTypes.TEXT] = location.query[JobFilterTypes.TEXT];
-      this.setState(state);
-    }
-  }
-
-  getHeadline(item, filteredJobs) {
-    const { state } = this;
-    const jobs = item.getItems();
-
-    const hasFiltersApplied = Object.keys(DEFAULT_FILTER_OPTIONS).some(
-      filterKey => {
-        return state[filterKey] != null && state[filterKey].length > 0;
-      }
-    );
 
-    if (hasFiltersApplied) {
-      return (
-        <FilterHeadline
-          onReset={this.resetFilter}
-          name="Jobs"
-          currentLength={filteredJobs.length}
-          totalLength={jobs.length}
-        />
-      );
-    }
-  }
-
-  getFilteredJobs(item) {
-    const { searchString } = this.state;
-    let jobs = item.getItems();
-
-    if (searchString) {
-      const filterProperties = Object.assign({}, item.getFilterProperties(), {
-        name(item) {
-          return item.getId();
-        }
-      });
-
-      jobs = item.filterItemsByText(searchString, filterProperties).getItems();
-    }
-
-    return jobs;
-  }
-
-  getJobTreeView(item, modal) {
-    const filteredJobs = this.getFilteredJobs(item);
-
-    return (
-      <Page>
-        <Page.Header
-          addButton={{
-            label: "Create a job",
-            onItemSelect: this.handleOpenJobFormModal
-          }}
-          breadcrumbs={<JobsBreadcrumbs tree={DCOSStore.jobTree} item={item} />}
-        />
-        <div className="flex-grow">
-          {this.getHeadline(item, filteredJobs)}
-          <FilterBar>
-            <JobSearchFilter
-              onChange={this.handleFilterChange}
-              value={this.state.searchString}
-            />
-          </FilterBar>
-          <JobsTable jobs={filteredJobs} />
-        </div>
-        {modal}
-      </Page>
-    );
-  }
-
-  getAlertPanelFooter() {
-    return (
-      <div className="button-collection flush-bottom">
-        <button
-          className="button button-primary"
-          onClick={this.handleOpenJobFormModal}
-        >
-          Create a Job
-        </button>
-      </div>
+    const modal = (
+      <JobFormModal
+        open={this.state.isJobFormModalOpen}
+        onClose={this.handleCloseJobFormModal}
+      />
     );
-  }
 
-  getContents(item, modal) {
     // Render loading screen
-    if (!DCOSStore.jobDataReceived) {
-      return (
-        <Page>
-          <Page.Header
-            breadcrumbs={<JobsBreadcrumbs tree={DCOSStore.jobTree} />}
-          />
-          <Loader />
-        </Page>
-      );
+    if (!isLoading) {
```

Comment:
We could move this into Container. Since it is in charge of the loading state. So that the container either shows `LoadingScreen` or `JobsTab` what do you think? And no need passing a boolean flag down